src/main/java/com/google/devtools/build/lib/syntax/MethodDescriptor.java

```diff
@@ -59,19 +62,157 @@ private MethodDescriptor(
       boolean allowReturnNones,
       boolean useStarlarkThread,
       boolean useStarlarkSemantics) {
+    boolean isStringModule = method.getDeclaringClass() == StringModule.class;
+
     this.method = method;
     this.annotation = annotation;
     this.name = name;
     this.doc = doc;
     this.documented = documented;
     this.structField = structField;
-    this.parameters = parameters;
+    this.parameters =
+        !isStringModule ? parameters : Arrays.copyOfRange(parameters, 1, parameters.length);
     this.extraPositionals = extraPositionals;
     this.extraKeywords = extraKeywords;
     this.selfCall = selfCall;
     this.allowReturnNones = allowReturnNones;
     this.useStarlarkThread = useStarlarkThread;
     this.useStarlarkSemantics = useStarlarkSemantics;
+    try {
+      MethodHandle mh = MethodHandles.lookup().unreflect(method);
```

Comment:
Save lookup() in a private static field? I don't think it's entirely free.

Reply:
Done.